README_cmake.md

```diff
@@ -1,106 +1,377 @@
 
-The Python package manager `pip` packages the newest version of CMake at all
-times. If you already have a Python 3 installation, this should be as easy as
-running
+```
+$ pip3 install -r .\python_bindings\requirements.txt
+```
+
+from the root of the repository.
+
+If you would like to use Ninja, note that it is installed alongside CMake when
+using the Visual Studio 2019 installer. Alternatively, you can install via
+Chocolatey or place the pre-built binary from their website in the PATH.
 
 ```
-$ pip install --upgrade cmake
+> choco install ninja
 ```
 
-See the [PyPI website](https://pypi.org/project/cmake) for more details.
+### macOS
 
-#### Windows
+On macOS, it is possible to install all dependencies via Homebrew:
 
-On Windows, you have two good options for installing CMake. First, if you have
-**Visual Studio 2019** installed, you can get CMake 3.16 through the Visual
-Studio installer. This is the recommended way of getting CMake if you are able
-to use Visual Studio 2019. See the
-[Visual C++ documentation](https://docs.microsoft.com/en-us/cpp/build/cmake-projects-in-visual-studio?view=vs-2019)
-for more details.
+```
+$ brew install llvm libpng libjpeg python@3.8 openblas doxygen ninja
+```
 
-**Otherwise**, you should not install CMake through Visual Studio, but rather
-from [Kitware's website](https://cmake.org/download/).
+The `llvm` package includes `clang`, `clang-format`, and `lld`, too.
 
-#### macOS
+### Ubuntu
 
-On macOS, a
-[recent version of CMake](https://formulae.brew.sh/cask/cmake#default) is
-readily available from [Homebrew](https://brew.sh). Simply running
+Finally, on Ubuntu 20.04 LTS, you should install the following packages:
 
 ```
-$ brew update
-$ brew install cmake
+dev@ubuntu:~$ sudo apt install \
+                  clang-tools lld llvm-dev libclang-dev liblld-10-dev \
+                  libpng-dev libjpeg-dev libgl-dev \
+                  python3-dev python3-numpy python3-scipy python3-imageio python3-pybind11 \
+                  libopenblas-dev libeigen3-dev libatlas-base-dev \
+                  doxygen ninja-build
 ```
 
-should install the newest version of CMake.
+# Building Halide with CMake
 
-The [releases on Kitware's website](https://cmake.org/download/) are also viable
-options.
+## Basic build
 
-#### Linux
+These instructions assume that your working directory is the Halide repo root.
 
-If you're on **Ubuntu Linux 20.04** (focal), then simply running
-`sudo apt install cmake` is sufficient.
+### Windows
 
-For older versions of **Debian, Ubuntu, Mint, and derivatives**, Kitware
-provides an [APT repository](https://apt.kitware.com/) with up-to-date releases.
-Note that this is still useful for Ubuntu 20.04 because it will remain up to
-date.
+If you plan to use the Ninja generator, be sure to be in the developer command
+prompt corresponding to your intended environment. In particular, if you wish to
+compile a 32-bit version of Halide, you should use the 64-bit tools. You can
```

Comment:
32 vs 64 bit mismatch in this sentence?

Reply:
Nope. I am about to push a clarification. Using the 32-bit host tools results in a linker crash.